File: OCaml/leapyear.ml
let is_leap_year year =
  (year mod 4 = 0 && year mod 100 <> 0) || year mod 400 = 0

let dates = [1900; 2000; 2023; 2024; 2400]

let () =
  List.iter (fun year ->
      Printf.printf "Is %d a leap year? %b\n" year (is_leap_year year)
    ) dates

File: Haskell/LeapYear.hs
isLeapYear :: Integer -> Bool
isLeapYear year
    | year `mod` 400 == 0 = True
    | year `mod` 100 == 0 = False
    | year `mod` 4 == 0   = True
    | otherwise           = False

formatResult :: Integer -> String
formatResult year = "Is " ++ show year ++ " a leap year? " ++ show (isLeapYear year)

main :: IO ()
main = do
    let years = [1900, 2000, 2023, 2024]
    mapM_ (putStrLn . formatResult) years